dev/io.openliberty.webcontainer.servlet.6.0.internal/src/io/openliberty/webcontainer60/srt/SRTServletRequest60.java

```diff
@@ -14,55 +14,57 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import jakarta.servlet.http.HttpServletRequest;
-import jakarta.servlet.ServletConnection;
-
 import com.ibm.websphere.ras.TraceComponent;
 import com.ibm.websphere.servlet.request.IRequest;
 import com.ibm.ws.webcontainer40.srt.SRTServletRequest40;
 import com.ibm.wsspi.webcontainer.logging.LoggerFactory;
 
 import io.openliberty.webcontainer60.osgi.srt.SRTConnectionContext60;
+import io.openliberty.websphere.servlet60.IRequest60;
+import jakarta.servlet.ServletConnection;
+import jakarta.servlet.http.HttpServletRequest;
 
 public class SRTServletRequest60 extends SRTServletRequest40 implements HttpServletRequest {
 
     protected static final Logger logger = LoggerFactory.getInstance().getLogger("io.openliberty.webcontainer60.srt");
     private static final String CLASS_NAME = SRTServletRequest60.class.getName();
-    
+
     private static long startID = 1;
     private static AtomicLong counter = new AtomicLong(startID);
 
     public SRTServletRequest60(SRTConnectionContext60 context) {
         super(context);
     }
-    
+
     @Override
     public void initForNextRequest(IRequest req) {
         String methodName = "initForNextRequest";
 
-        if (TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
-            logger.logp(Level.FINE, CLASS_NAME, methodName, "this [" + this + "] , req [" + req +"]");
-        }
         super.initForNextRequest(req);
-        
+
         if (req != null) {
             super.setSrtRequestId(String.valueOf(counter.getAndIncrement()));
 
+            this.initForServletConnection();
+
             if (TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
-                logger.logp(Level.FINE, CLASS_NAME, methodName, "this [" + this + "] , requestID [" + this.getRequestId() +"]");
+                logger.logp(Level.FINE, CLASS_NAME, methodName,
+                            "this [" + this + "] , request id [" + this.getRequestId() + "] , connection id ["
+                                                                + ((SRTServletConnection) super.getSrtServletConnection()).getConnectionId() + "] , req [" + req + "]");
             }
-            
-            this.setupServletConnection();
         }
     }
-    
-    private void setupServletConnection() {
+
+    private void initForServletConnection() {
         SRTServletConnection servletConn = new SRTServletConnection();
-        servletConn.setConnectionID(this.getRequestId());
-        
+
+        servletConn.setConnectionID(String.valueOf(((IRequest60) _request).getConnectionId()));
+        servletConn.setConnectionSecure(super._request.isSSL());
+        servletConn.setProtocol(super._request.getProtocol());
```

Comment:
The API where this is used is ServletConnection.getProtocol: https://jakarta.ee/specifications/servlet/6.0/apidocs/jakarta.servlet/jakarta/servlet/servletconnection#getProtocol()

It doesn't sound like ServletConnection.getProtocol should return the same things as ServletRequest.getProtocol

Reply:
That's my understanding as this point.  It is the protocol after we strip off all the secure wrapper protocol (like SSL, TLS...)  It is the protocol that the server received for a request.

Reply:
According to the API, the identification sequence should be returned in form of a String

https://www.iana.org/assignments/tls-extensiontype-values/tls-extensiontype-values.xhtml#alpn-protocol-ids

Example: HTTP/1.1

HTTP/1.1 | 0x68 0x74 0x74 0x70 0x2f 0x31 0x2e 0x31 ("http/1.1")
-- | --

We will return the protocol (HTTP/1.1) for now.